core/src/main/scala/kafka/server/BrokerServer.scala

```diff
@@ -272,7 +273,7 @@ class BrokerServer(
       val networkListeners = new ListenerCollection()
       config.advertisedListeners.foreach { ep =>
         networkListeners.add(new Listener().
-          setHost(ep.host).
+          setHost(if (Utils.isBlank(ep.host)) InetAddress.getLocalHost.getCanonicalHostName else ep.host).
```

Comment:
I feel it is ok to check blank manually. It won't bring huge code conflicts.

Reply:
> I feel it is ok to check blank manually. It won't bring huge code conflicts.

Good suggestion.Submit your feedback. :)